### DISCUSSION TITLE
Just Would Like To Say Thanks

### DISCUSSION THREAD

**Discussion by dtquocbao** [oqtane/oqtane.framework#963] [Show and tell]
Finally, I would like to show my pet project which developed using Oqtane (version 2.0, .Net5, WebAssembly). This is an application to support clinic system which still under developing. However, with this current version, we're proudly to support basic requirements such as 
1. Manage patients(basic information/contact, sign on Consent forms), 
2. Manage treatment records, 
3. Manage Physical exams, 
4. Billing history (based on ICD10 or CPT code items).... 

Most of this features can be managed by end-user with administrator permission. 

Obviously, we're still fixing bugs and doing some enhancements e.g.:

1. Update new theme by using sb-admin-pro theme
2. Apply Syncfusion  blazor components also apply Signature Pad components (current version not work with latest Blazor version and unfortunately it's not a priority to Syncfusion team)
3. Add Scheduling/Appointment feature 
4. Try to add Video streaming feature even Video chat feature
...and so many more

So I am showing my many respects and appreciations to @sbwalker as well as other contributors who make .NET developer life easier! 

Happy Coding Folks!!!!

![image](https://user-images.githubusercontent.com/4122092/100599809-561ad300-32b5-11eb-8123-302d787e2253.png)

**Comment by justSteve** [comment]
Congrats on your milestone Rocky. I'm exploring the use of Oqtane for a similar Line of Business -- exercise studios instead of clinics but things like user management, billing, and scheduling domains would be similar to what you've had to manage.

I'd be very interested in how you've gone about structuring your modules. I'd presume you've created at least one per 'area of concern'. 

thanks much for anything you are able to share.

**Reply by dtquocbao** [reply]
Thanks @justSteve,

Yeah, as I mentioned that one of our next targets is Video Streaming which sounds similar to "online therapy exercises" using our tracking devices. 

Currently I organize modules by kind of mixing features and roles. For example I created some pages which cover the area of group modules. Then I created modules and grant the permission to certain roles so users of those roles can share those modules. 

Two things that I extremely love Oqtane are: 1. Necessary stuffs such as emails/settings help to cover my infrastructure tasks. 2. Also basic CRUD of Module Creator supports me to create back-end modules like configurations to Admin role users so I almost took nothing of my effort to develop back-end settings, configurations and save me lots of time. So I can 90% focus on main features and business.

I have never developed an Asp.net app within less than a month with those kind of feature as well as UI design but it took me 1.5 months to get this version of the software while I was trying to learn Blazor stuff.

That's insane to me!

**Reply by justSteve** [reply]
Thanks for the reply -- would it be fair to say your top level modules consists of roles (groups?) are 'Admin' and 'User' and features are controlled by how the permissions are granted?

**Comment by sbwalker** [comment]
Thank you @rockydant for sharing! It is great to see the creative implementations of Oqtane that the community creates!

**Reply by justSteve** [reply]
@sbwalker I'm very curious as to your thoughts on how a project like @rockydant's is best structured. It's a very common requirement where a set of services like notifications, user management, admin functions, calendaring and so on. Pretty much the same core services to different lines of business. 

Probably there are examples of DNN efforts that are published as jumpstarters or generic LOB apps. Are there comparisons between how a DNN would structure modules vs. Oqtane? 

If there's any value in reviewing DNN samples toward understanding app structure I'd love any links that come to mind.

**Comment by hishamco** [comment]
Horary, great you build something awesome with Oqtane @rockydant, please let us know if you face some issues or limitations that may help the community and to make Oqtane development easier
